frontend/src/views/ExistingParticipantView.js

```diff
@@ -41,25 +41,30 @@ const ExistingParticipantView = observer(() => {
   const participantStore = rootStore.ParticipantStore
   // Utility store derived from root store
   const utilityStore = rootStore.UtilityStore
-  // set up history for routing pushes
+
+  const { participantId } = useParams()
   const history = useHistory()
 
   const existingParticipant = toJS(participantStore.participant)
   const insurers = toJS(participantStore.insurers)
 
   useEffect(() => {
-    // kick off api calls for insurance list from Mobx
+    // kick off api calls for relevant data
     participantStore.getInsurers()
-    // kick off api calls for program list from Mobx
     participantStore.getPrograms()
-    // -----------------------------------
-
-    // return to /participants page if current participant is not verified with an id
-    // add participants in NewParticipantView
-    if (!participantStore.participant.id) {
-      // todo: reset to empty  object?
-      history.push("/participants")
+    async function matchParticpantIdToParams() {
+      if (participantStore.participant.id !== parseInt(participantId, 10)) {
+        // get participant data via param id if not in state or mismatch between state and route
+        const ok = await participantStore.getParticipant(participantId)
+        if (!ok) {
+          participantStore.setDefaultParticipant()
+          participantStore.setDefaultVisit()
+          history.push("/404/")
+        }
+      }
     }
+
+    matchParticpantIdToParams()
```

Comment:
Where is this function defined?

Reply:
within the useEffect block itself. the callback given to useeffect cant be async